pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,216 @@
+package latencyprofilecontroller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	v1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	listerv1 "github.com/openshift/client-go/config/listers/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	controlplanev1 "github.com/openshift/api/kubecontrolplane/v1"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+)
+
+const (
+	operatorNamespace = operatorclient.OperatorNamespace
+	targetNamespace   = operatorclient.TargetNamespace
+
+	ApiServerConfigMapName = "config"
+	ApiServerConfigMapKey  = "config.yaml"
+
+	DefaultNotReadyTolerationSecondsArgument    = "default-not-ready-toleration-seconds"
+	DefaultUnreachableTolerationSecondsArgument = "default-unreachable-toleration-seconds"
+)
+
+// LatencyProfileController
+type LatencyProfileController struct {
+	operatorClient  v1helpers.StaticPodOperatorClient
+	configMapClient corev1client.ConfigMapsGetter
+	nodeLister      listerv1.NodeLister
+	nodeClient      v1.NodeInterface
+}
+
+func NewLatencyProfileController(
+	operatorClient v1helpers.StaticPodOperatorClient,
+	configClient configv1client.Clientset,
+	configInformer configinformers.SharedInformerFactory,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	kubeClient kubernetes.Interface,
+	eventRecorder events.Recorder,
+) factory.Controller {
+
+	ret := &LatencyProfileController{
+		operatorClient:  operatorClient,
+		configMapClient: v1helpers.CachedConfigMapGetter(kubeClient.CoreV1(), kubeInformersForNamespaces),
+		nodeLister:      configInformer.Config().V1().Nodes().Lister(),
+		nodeClient:      configClient.ConfigV1().Nodes(),
+	}
+	configClient.ConfigV1().Nodes()
+
+	return factory.New().WithInformers(
+		kubeInformersForNamespaces.InformersFor(targetNamespace).Core().V1().ConfigMaps().Informer(),
+		operatorClient.Informer(),
+		configInformer.Config().V1().Nodes().Informer(),
+	).ResyncEvery(time.Second).WithSync(ret.sync).ToController("LatencyProfileController", eventRecorder)
```

Comment:
why not to call `c.ensureProfileSyncedToAllNodes` directly?

Reply:
Done, renamed `c.ensureProfileSyncedToAllNodes` to `c.sync` and used it within `.WithSync`; so we don't have an extraneous function now. 

IMHO, `ensureProfileSyncedToAllNodes` was incorrectly named as it was more like a function to: update config node object status on the basis of whether all node revisions have the specific argument rolled out or not.